### PR TITLE
EZ_23  Integrated github issue list in todo format

### LINKED ISSUE #23: Integrate github issue list

### PR DESCRIPTION
**Pull request by CorvusChrome**

### CODE REVIEW

File: webapp/package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.5.0",
     "@testing-library/user-event": "^7.2.1",
+    "prop-types": "^15.7.2",
```
Comment (randomdima) [inline webapp/package.json:9]: Why do you need this package for?

Comment (CorvusChrome) [reply webapp/package.json:9]: For setting required variables types

File: webapp/src/index.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ import App from './App';
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <App/>
```
Comment (randomdima) [inline webapp/src/index.js:8]: Try to not commit unnecessary changes like this.

File: webapp/src/Todo/TodoList.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import TodoItem from './TodoItem'
+import PropTypes from 'prop-types'
+
+
+function TodoList(prop) {
```
Comment (randomdima) [inline webapp/src/Todo/TodoList.js]: Can you deconstruct args here the same way as in TodoItem

File: webapp/src/Todo/TodoList.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    return (
+        <ul>
+            {prop.todos.map((todo, index) => {
```
Comment (randomdima) [inline webapp/src/Todo/TodoList.js]: Lambda can be simplified.
No need to use return here.

File: webapp/src/Todo/TodoItem.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import React, { useContext } from 'react'
+import PropTypes from 'prop-types'
+import Context from '../context'
+
+function TodoItem({ todo, index, onChange }) {
+    const { removeTodo } = useContext(Context)
+    const classes = []
```
Comment (randomdima) [inline webapp/src/Todo/TodoItem.js]: You only have a single class to use - no need to create an array for this.

File: webapp/src/Todo/TodoItem.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+                {index + 1}&nbsp;{todo.title}
+            </span>
+            <button onClick={() => removeTodo(todo.id)}>&times;</button>
+        </li>
+    )
+}
+
+TodoItem.propTypes = {
+    todo: PropTypes.object.isRequired,
+    index: PropTypes.number,
```
Comment (randomdima) [inline webapp/src/Todo/TodoItem.js]: Index looks like required property from how it is being used.

File: webapp/src/Todo/TodoItem.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import React, { useContext } from 'react'
+import PropTypes from 'prop-types'
+import Context from '../context'
+
+function TodoItem({ todo, index, onChange }) {
+    const { removeTodo } = useContext(Context)
```
Comment (randomdima) [inline webapp/src/Todo/TodoItem.js]: Why one button click is an argument and the other is in context?

File: webapp/src/Todo/AddTodo.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+
+
+const styles = {
```
Comment (randomdima) [inline webapp/src/Todo/AddTodo.js]: Why you  did not put styles in css?

Comment (CorvusChrome) [reply webapp/src/Todo/AddTodo.js]: im thinking about to put all styles into js files. 
Found this more comfortable to use with bigger projects. And keep css only for common cases..

File: webapp/src/App.js [hunk 9]
```diff
@@ -1,11 +1,51 @@
-import React from 'react';
+import React, { useState } from 'react';
+import TodoList from './Todo/TodoList'
+import Context from './context'
+import AddTodo from './Todo/AddTodo';
 
 function App() {
+  const [todos, setTodos] = React.useState([  ])
+
+  useState(() => {
```
Comment (randomdima) [inline webapp/src/App.js:8]: This does not look like a state from the way you use it.
Without useStage it works just the same.

File: webapp/src/App.js [hunk 10]
```diff
@@ -1,11 +1,51 @@
...
+    fetch('https://api.github.com/repos/randomdima/ez/issues')
+      .then(response => response.json())
+      .then(todos => {
+        setTodos(todos)
+      })
+  }, [])
+
+  function toggoleTodo(id) {
```
Comment (randomdima) [inline webapp/src/App.js]: typo

File: webapp/src/App.js [hunk 11]
```diff
@@ -1,11 +1,51 @@
...
+    setTodos(todos.map(todo => {
+      if (todo.id === id) {
+        todo.completed = !todo.completed
+      }
+      return todo
+    }))
+  }
+  function addTodo(title) {
+    setTodos(todos.concat([{
```
Comment (randomdima) [inline webapp/src/App.js]: You dont need to pass array into contact if you want to add single element.

File: webapp/src/App.js [hunk 12]
```diff
@@ -1,11 +1,51 @@
...
+import AddTodo from './Todo/AddTodo';
 
 function App() {
+  const [todos, setTodos] = React.useState([  ])
+
+  useState(() => {
+    fetch('https://api.github.com/repos/randomdima/ez/issues')
+      .then(response => response.json())
+      .then(todos => {
+        setTodos(todos)
```
Comment (randomdima) [inline webapp/src/App.js]: Lambda like this can be simplified to `then(setTodos)`

File: webapp/src/index.css [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+    font-size: 68px;
+}
+
+p {
+    padding-top: 1rem;
+    margin: 0 auto;
+    text-align: center;
+    font-family: cursive;
+    font-size: 30px;
+    margin: 0;
```
Comment (randomdima) [inline webapp/src/index.css]: margin twice

File: webapp/src/App.js (hunk 10 above)
Comment (CorvusChrome) [reply webapp/src/App.js]: sam ty typoi :0

**Review by randomdima — DISMISSED** [review]

**Review by randomdima — APPROVED** [review]
